From 77ba264dfa17925c31a39f98b4ab0010dc2e7651 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@nxp.com>
Date: Mon, 16 Jul 2018 09:22:20 -0500
Subject: [PATCH 4331/5242] MLK-18904-9: arch: ARM64: dts: Adding DisplayPort
 support for i.MX8M (MCU2)

commit  7aefea3d7af26ecb30c5d296a4dd1bf01d2c3b10 from
https://source.codeaurora.org/external/imx/linux-imx.git

Adding support for DisplayPort and embedded DisplayPort.

Signed-off-by: Oliver Brown <oliver.brown@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    4 +++-
 .../arm64/boot/dts/freescale/fsl-imx8mq-evk-dp.dts |   22 +++++++++++++++++
 .../boot/dts/freescale/fsl-imx8mq-evk-edp.dts      |   25 ++++++++++++++++++++
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi      |    3 ++-
 4 files changed, 52 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dp.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-edp.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 26d25c2..ce79cad 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -93,7 +93,9 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-ddr3l-arm2.dtb \
 				 fsl-imx8mq-evk-audio-tdm.dtb \
 				 fsl-imx8mq-evk-drm.dtb \
 				 fsl-imx8mq-evk-root.dtb \
-				 fsl-imx8mq-evk-inmate.dtb
+				 fsl-imx8mq-evk-inmate.dtb \
+				 fsl-imx8mq-evk-dp.dtb \
+				 fsl-imx8mq-evk-edp.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MM) += fsl-imx8mm-evk.dtb \
 				 fsl-imx8mm-evk-ak4497.dtb \
 				 fsl-imx8mm-evk-m4.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dp.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dp.dts
new file mode 100644
index 0000000..057c0e6
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dp.dts
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mq-evk.dts"
+
+&hdmi {
+	compatible = "fsl,imx8mq-dp";
+	lane_mapping = <0xc6>;
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-edp.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-edp.dts
new file mode 100644
index 0000000..5f8ea76
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-edp.dts
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mq-evk.dts"
+
+&hdmi {
+	compatible = "fsl,imx8mq-dp";
+	lane_mapping = <0xc6>;
+	fsl,edp;
+	edp_link_rate = <0x6>;
+	edp_num_lanes = <0x4>;
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 5104c8e..dd287c4 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -590,7 +590,8 @@
 		#size-cells = <0>;
 		compatible = "fsl,imx8mq-hdmi";
 		reg = <0x0 0x32c00000 0x0 0x100000>,	/* HDP registers */
-				<0x0 0x32e40000 0x0 0x40000>;   /* HDP SEC register */
+				<0x0 0x32e40000 0x0 0x40000>, /* HDP SEC register */
+				<0x0 0x32e2f000 0x0 0x10>;    /* RESET register */
 		interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>,
 					<GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "plug_in", "plug_out";
-- 
1.7.9.5

